core/src/test/java/org/apache/iceberg/TestOverwriteWithValidation.java

```diff
@@ -333,6 +333,31 @@ public void testOverwriteCompatibleAdditionStrictValidated() {
         committedSnapshotId, table.currentSnapshot().snapshotId());
   }
 
+  @Test
+  public void testOverwriteCompatibleAdditionStrictValidatedNoConflict() {
```

Comment:
Well, I am not sure I understand this completely.

Under the current implementation, this test would throw a validation exception as the conflict detection filter is `true`. That means any concurrent modification should be considered a conflict. I think that's the correct behavior. It will happen if a row-level operation did not have a predicate that could be pushed down. For instance, `UPDATE t SET col = 1`. Any append should be considered a conflict in this case. Otherwise, suppose we had only files in partition B and then got an UPDATE statement without a condition. While executing the UPDATE statement, someone concurrently added new files to partition A. The UPDATE will be deleting files from partition B but the concurrently added files to partition A should still fail the operation.

@coolderli, will adding a partition predicate to you MERGE statement solve the issue? You can provide a predicate in the ON condition and that will be used as the conflict detection filter.

Reply:
@aokolnychyi Thanks, I think a partition predicate will work. But I worry about that because using the hidden partition we may not know which partition should be filtered. If we use a partition like `bucket`, I don't know how to express this predicate.

Reply:
Maybe we should use other partitions, but this brings more difficulties for users to write SQL. Users must be concerned about the partition distribution of data. In the above example, the files of partition A and partition B will not conflict, but they will still conflict at present because there is no partition predicate in SQL. Users may feel confused because conflict resolution should be the concern of iceberg, not users. I means if concurrent submission does not conflict, we should allow submission.

Reply:
This is an area where we're really like for Spark to be able to support transform expressions in pushdown. Iceberg already supports expressions like `day(ts) = '2022-01-01'` and that would be really convenient for some use cases.

Until then, one thing that we can do is to start accepting predicates from Spark for the `_partition` metadata column. In Spark 3.2, we've added a metadata column that returns the partition for a given row. I think that we could recognize predicates for those columns and convert them into expressions. For example, we could take `_partition.id_bucket=55` and convert it to `bucket(256, id) = 55` and push that down to cover the use case you're trying to solve here.

Would you like to try implementing this, @coolderli?

Reply:
@rdblue Yes, I think it is useful, I will take a look.